ctpolicy/ctpolicy.go

```diff
@@ -124,6 +127,7 @@ func (ctp *CTPolicy) GetSCTs(ctx context.Context, cert core.CertDER, expiration
 		select {
 		case <-subCtx.Done():
 			return nil, "", subCtx.Err()
+		case <-unblocker:
```

Comment:
Unfortunately, this doesn't quite do what we want. We kick off a bunch of `getOne` goroutines, and each one controls its own sleep duration based on the `i` parameter that gets passed in. This would cause a _random_ one of the blocked goroutines to unblock, not necessarily the "next" (i.e. closest to finishing its sleep anyway) one. In the worst case, this could cause two goroutines to wake up simultaneously: one from its timer, the other from this channel. That's... not a terrible case, but it's sufficiently surprising/confusing that I'd prefer a different approach.

This would be pretty easy if the loop on line 163 was in its own background goroutine, and it centrally managed the sleeps between kicking off getOne goroutines. But that's a larger refactoring that you may not be interested in tackling.

Reply:
I don't think unblocking a random goroutine is the worst thing ever, though I definitely more worry about this logic getting more complicated and hard-to-follow.  I think my feelings are that if we want this functionality, we should probably refactor this -- and you're right, I don't want to do that right now.